File: practicum-java/src/main/java/ru/practicum/kafka/streams/model/BlockCommand.java
package ru.practicum.kafka.streams.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Команда изменения списка блокировок — запись топика blocked_users.
 *
 * <p>В топик кладутся именно команды, а не готовый список: чтобы заблокировать
 * кого-то, клиенту достаточно знать одну пару «кто кого», а не весь текущий список
 * пользователя. Из потока команд топология сворачивает актуальное множество
 * (см. MessagingTopology), так что состояние восстанавливается проигрыванием топика
 * с начала.
 *
 * <p>Ключ записи Kafka — {@code blocker_id}: он же ключ таблицы блокировок,
 * и именно по нему поток сообщений совмещается с ней без repartition.
 */
public record BlockCommand(
        @JsonProperty("blocker_id") String blockerId,
        @JsonProperty("blocked_id") String blockedId,
        @JsonProperty("action") String action) {

    public static final String ACTION_BLOCK = "block";
    public static final String ACTION_UNBLOCK = "unblock";

    public boolean isBlock() {
        return ACTION_BLOCK.equalsIgnoreCase(action);
    }
}

File: practicum-dotnet/src/Practicum.Kafka.Streams/Models/BlockCommand.cs
using System.Text.Json.Serialization;

namespace Practicum.Kafka.Streams.Models;

/// <summary>
/// Команда изменения списка блокировок — запись топика blocked_users.
///
/// В топик кладутся именно команды, а не готовый список: чтобы заблокировать
/// кого-то, клиенту достаточно знать одну пару «кто кого», а не весь текущий
/// список пользователя. Из потока команд топология сворачивает актуальное
/// множество, так что состояние восстанавливается проигрыванием топика с начала.
///
/// Ключ записи Kafka — blocker_id: он же ключ таблицы блокировок, и именно
/// по нему поток сообщений совмещается с ней без repartition.
/// </summary>
public sealed record BlockCommand
{
    public const string ActionBlock = "block";

    [JsonPropertyName("blocker_id")]
    public string BlockerId { get; init; } = string.Empty;

    [JsonPropertyName("blocked_id")]
    public string BlockedId { get; init; } = string.Empty;

    [JsonPropertyName("action")]
    public string Action { get; init; } = string.Empty;

    [JsonIgnore]
    public bool IsBlock => string.Equals(Action, ActionBlock, StringComparison.OrdinalIgnoreCase);
}
